File: yabo-shop-api/src/main/java/uz/kvikk/yabo/utils/converter/DateConverter.java
package uz.kvikk.yabo.utils.converter;

import org.jooq.impl.AbstractConverter;

import java.sql.Date;

public class  DateConverter extends AbstractConverter<Date, java.util.Date> {

    private static final long serialVersionUID = -6914493125390333501L;

    public DateConverter() {
        super(Date.class, java.util.Date.class);
    }

    @Override
    public final java.util.Date from(Date t) {
        return t == null ? null : new java.util.Date(t.getTime());
    }

    @Override
    public final Date to(java.util.Date u) {
        return u == null ? null : new Date(u.getTime());
    }
}

File: file-storage-api/src/main/java/uz/kvikk/file/utils/converter/DateConverter.kt
package uz.kvikk.file.utils.converter

import org.jooq.impl.AbstractConverter

import java.sql.Date

class DateConverter : AbstractConverter<Date, java.util.Date>(Date::class.java, java.util.Date::class.java) {

  override fun from(t: Date?): java.util.Date? {
    return if (t == null) null else java.util.Date(t.time)
  }

  override fun to(u: java.util.Date?): Date? {
    return if (u == null) null else Date(u.time)
  }
}

